### PR TITLE
Bump matrix-sdk-crypto-wasm to 5.0.0

### PR DESCRIPTION
**Pull request by richvdh**
Slightly more involved than normal because it requires us to pass a backup version into `OlmMachine.importBackedUpRoomKeys`.

On the other hand we can now re-enable the test that was disabled in https://github.com/matrix-org/matrix-js-sdk/pull/4214 due to https://github.com/matrix-org/matrix-rust-sdk/issues/3447

Fixes: https://github.com/element-hq/element-web/issues/27165

element-web notes: Fix a bug in the Rust crypto implementation which caused repeated HTTP requests to `/keys/query` when the session was unverified.

### CODE REVIEW

File: src/crypto/index.ts [hunk 1]
```diff
@@ -1888,7 +1888,11 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
     /**
      * Implementation of {@link CryptoBackend#importBackedUpRoomKeys}.
      */
-    public importBackedUpRoomKeys(keys: IMegolmSessionData[], opts: ImportRoomKeysOpts = {}): Promise<void> {
+    public importBackedUpRoomKeys(
```
Comment (andybalaam) [inline src/crypto/index.ts:1891]: Does this change a public interface and do we need to do something if so?

**Review by andybalaam — APPROVED** [review]
Looks good; one question.

File: src/crypto/index.ts (hunk 1 above)
Comment (richvdh) [reply src/crypto/index.ts:1891]: no, CryptoBackend is an internal interface.

Comment (richvdh) [reply src/crypto/index.ts:1891]: (and legacy `Crypto` is documented as being an internal interface, even though we weren't very good at following that.)

**Comment by richvdh** [comment]
The coverage gate is complaining about legacy crypto. I'm afraid I'm not going to fix that.

**Pull request merged**